channeldb/payment_control.go

```diff
@@ -170,10 +171,42 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 
 		// We can only register attempts for payments that are
 		// in-flight.
-		if err := ensureInFlight(bucket); err != nil {
+		paymentStatus, terminal, sentAmt, err := fetchPaymentStatus(
```

Comment:
is it the case that over rpc we already translate the settled terminal case into the an early success? i believe so, but wanted to double check

Reply:
No, the RPC reflects the DB level status now.

I think the real fix for realtime updates here is to update the `TrackPayment` RPC with HTLC notifications (which is planned).

Reply:
Isn't that a breaking change? I too had the impression that we'd do the mapping to final states in this pr

Reply:
It's not breaking, since this is new behavior. A payment couldn't be both settled and still in flight before MPP.

We can redefine the definition on the RPC level, if we think it's a big benefit. Only thing is I see it become a little more complex to handle two different definitions in the codebase.

Reply:
Then if it is not breaking now, we should definitely define it right. Otherwise we can't change it anymore after the 0.10 release?